samcli/local/apigw/local_apigw_service.py

```diff
@@ -375,8 +375,20 @@ def _request_handler(self, **kwargs):
             LOG.error("Invalid lambda response received: %s", ex)
             return ServiceErrorResponses.lambda_failure_response()
 
+        self._add_cors_headers_to_response(method, headers, cors_headers)
         return self.service_response(body, headers, status_code)
 
+    @staticmethod
+    def _add_cors_headers_to_response(method, response_headers, cors_headers):
+        # If the CORS headers are present and one of the allowed methods is being used,
+        # add the allow-origin header to the response.
+        # On Origin other than the wildcard (*), include Vary: Origin
+        allow_origin = cors_headers.get("Access-Control-Allow-Origin")
+        if method in cors_headers.get("Access-Control-Allow-Methods", []) and allow_origin:
+            response_headers.add_header("Access-Control-Allow-Origin", allow_origin)
```

Comment:
According to the API Gateway docs, the Lambda function is responsible for adding these integrations: https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-cors.html (in the "Enabling CORS support for Lambda or HTTP proxy integrations" section). So SAM CLI should not be doing this.